text/src/autogluon/text/automm/data/preprocess_dataframe.py

```diff
@@ -175,8 +175,9 @@ def fit(
             elif col_type == CATEGORICAL:
                 if self._config["categorical"]["convert_to_text"]:
                     # Convert categorical column as text column
-                    processed_data = col_value.apply(lambda ele: '' if ele is None else str(ele))
-                    if len(np.unique(processed_data)) == 1:
+                    col_value = col_value.astype("object")
```

Comment:
We may mark a TODO here. We need to investigate if this is the optimal choice.

Reply:
Basically, we can consider to treat it as categorical value if the column has the "category" type in pandas. However, this may require some further investigation so we may mark it with `#TODO(...) ...`

Reply:
If self._config["categorical"]["convert_to_text"] is True, we should convert categorical data to text since it is what the config requires. Users can set it False if not wanting this conversion.

Reply:
We keep categorical values to text if those categorical labels has meanings. We may mark a TODO about how we can do it better.

Reply:
I believe the feature generator is already handling this logic. We should consider align it in the future.

Reply:
Yes, considering the current time constraint, we may need to align the API after 0.4 release.

Reply:
I think we may pass the column types generated via tabular to the text in the integration: https://github.com/awslabs/autogluon/blob/c0e33c76b62db887684fe9b607b34420d6bb9b1e/tabular/src/autogluon/tabular/models/text_prediction/text_prediction_v1_model.py#L144-L150. It will guarantee the consistency.

Reply:
We can use the `feature_metadata` https://github.com/awslabs/autogluon/blob/c0e33c76b62db887684fe9b607b34420d6bb9b1e/tabular/src/autogluon/tabular/models/text_prediction/text_prediction_v1_model.py#L53